tsdoc/src/nodes/DocMemberSelector.ts

```diff
@@ -0,0 +1,105 @@
+import { DocNodeKind, DocNode } from './DocNode';
+import { DocNodeLeaf, IDocNodeLeafParameters } from './DocNodeLeaf';
+import { Excerpt } from '../parser/Excerpt';
+import { DocParticle } from './DocParticle';
+import { StringChecks } from '../parser/StringChecks';
+
+/**
+ * Constructor parameters for {@link DocMemberSelector}.
+ */
+export interface IDocMemberSelectorParameters extends IDocNodeLeafParameters {
+  labelExcerpt?: Excerpt;
+  label: string;
+}
+
+/**
+ * Kinds of TSDoc selectors.
+ */
+export const enum SelectorKind {
+  /**
+   * Used in cases where the parser encounters a string that is incorrect but
+   * valid enough that a DocMemberSelector node was created.
+   */
+  Error = 'error',
+
+  /**
+   * System selectors are always all lower-case and belong to a set of predefined label names.
+   */
+  System = 'system',
+
+  /**
+   * Index selectors are integer numbers.  They provide an alternative way of referencing
+   * overloaded functions, based on the order in which the declarations appear in
+   * a source file.
+   *
+   * @remarks
+   * Warning:  Index selectors are not recommended; they are intended to provide a temporary
+   * workaround for situations where an external library neglected to declare a `{@label}` tag
+   * and cannot be easily fixed.
+   */
+  Index = 'index',
+
+  /**
+   * Custom selectors refer to labels created using the `{@label}` TSDoc tag.
+   * The labels are always comprised of upper-case letters or numbers separated by underscores,
+   * and the first character cannot be a number.
+   */
+  Custom = 'custom'
```

Comment:
>custom [](start = 12, length = 6)

Maybe call this label.

Reply:
Good call

---
In reply to: [221409976](https://github.com/Microsoft/tsdoc/pull/75#discussion_r221409976) [](ancestors = 221409976)